test/acme_server/nonce_test.exs

```diff
@@ -0,0 +1,20 @@
+defmodule AcmeServer.NonceTest do
+  use ExUnit.Case, async: true
+
+  test "new" do
+    first_nonce = AcmeServer.Nonce.new()
+    nonce = AcmeServer.Nonce.new()
+    assert nonce == first_nonce + 1
+  end
+
+  test "verify ok" do
+    nonce = AcmeServer.Nonce.new()
+    assert :ok == AcmeServer.Nonce.verify!(nonce)
+  end
+
+  test "verify not ok" do
```

Comment:
I think a test verifying that a repeated verification of nonce fails would also be nice here.